Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/checkins/CheckInsViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package de.rki.coronawarnapp.ui.eventregistration.attendee.checkins
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.SavedStateHandle
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.eventregistration.checkins.qrcode.QRCodeUriParser
+import de.rki.coronawarnapp.eventregistration.checkins.qrcode.QRCodeVerifier
+import de.rki.coronawarnapp.eventregistration.checkins.qrcode.QRCodeVerifyResult
+import de.rki.coronawarnapp.exception.ExceptionCategory
+import de.rki.coronawarnapp.exception.reporting.report
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
+import timber.log.Timber
+
+class CheckInsViewModel @AssistedInject constructor(
+    @Assisted private val savedState: SavedStateHandle,
+    @Assisted private val deepLink: String?,
+    dispatcherProvider: DispatcherProvider,
+    private val qrCodeVerifier: QRCodeVerifier,
+    private val qrCodeUriParser: QRCodeUriParser
+) : CWAViewModel(dispatcherProvider) {
+
+    private val verifyResultData = MutableLiveData<QRCodeVerifyResult>()
+    val verifyResult: LiveData<QRCodeVerifyResult> = verifyResultData
+
+    init {
+        deepLink?.let {
+            if (deepLink != savedState.get(SKEY_LAST_DEEPLINK)) {
+                Timber.i("New deeplink: %s", deepLink)
+                verifyUri(it)
+            } else {
+                Timber.d("Already consumed this deeplink: %s", deepLink)
+            }
+        }
+        savedState.set(SKEY_LAST_DEEPLINK, deepLink)
```

Comment:
Do we need to save it? or it is better to clear it ?

Reply:
We need to save it in case the system restores the app after "don't keep activities state". The previously solution you wrote using "clear" on the arguments bundle works too, in most cases I guess, but this is not documented behavior, it just happens that on most devices the same bundle is reused, but theoretically the original bundle (not the one we called  "clear" on, could be returned to us.